tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -188,28 +187,23 @@ LogicalResult ConvertTFLReluOp::matchAndRewrite(
             .dyn_cast<mlir::quant::UniformQuantizedType>();
 
     Value op1_rescale_in =
-        buildRescaleToInt32(rewriter, op, tfl_relu_op.x(),
-                            input_qtype.getScale() / output_qtype.getScale(),
-                            input_qtype.getZeroPoint());
-    auto op2_relun_op1 = rewriter.create<tosa::ReluNOp>(
-        op->getLoc(), rescale_type, op1_rescale_in,
+        buildRescale(rewriter, op, output_type, tfl_relu_op.x(),
+                     input_qtype.getScale() / output_qtype.getScale(),
+                     input_qtype.getZeroPoint(), output_qtype.getZeroPoint(),
+                     /*double_round=*/false, /*scale32=*/true);
+    rewriter.replaceOpWithNewOp<tosa::ClampOp>(
+        op, output_type, op1_rescale_in,
+        rewriter.getI64IntegerAttr(output_qtype.getZeroPoint()),
         rewriter.getI64IntegerAttr(std::numeric_limits<int32_t>::max()),
-        rewriter.getF32FloatAttr(0.0f));
-    Value op3_rescale_op2 = buildRescaleFromInt32(
-        rewriter, op, output_type, op2_relun_op1.getResult(), 1.0f,
-        output_qtype.getZeroPoint());
-
-    output = op3_rescale_op2;
+        rewriter.getF32FloatAttr(0.0f), rewriter.getF32FloatAttr(0.0f));
   } else {
-    auto op1_relun_in = rewriter.create<tosa::ReluNOp>(
-        op->getLoc(), output_type, tfl_relu_op.x(),
-        rewriter.getI64IntegerAttr(0),
+    rewriter.replaceOpWithNewOp<tosa::ClampOp>(
```

Comment:
See comment below by limiting the branch. Specifically limit the branch to computing the min/max values and add the rescale for quantized case.

Reply:
Done.